repeater/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.repeater/Resources.scala

```diff
@@ -71,92 +75,96 @@ class Resources[F[_]: Sync](
 }
 
 object Resources {
+  private val QueueSize = 100
 
-  val QueueSize = 100
-
-  case class Statistics(inserted: Int, desperates: Int, lifetime: Duration)
-
+  final case class Statistics(inserted: Int, uninsertable: Int, lifetime: Duration)
   object Statistics {
-    val start = Statistics(0, 0, Duration(0, "millis"))
+    val start: Statistics = Statistics(0, 0, Duration(0, "millis"))
 
     implicit val showRepeaterStatistics: Show[Statistics] = {
       case s if (s.lifetime.toHours == 0) =>
-        s"Statistics: ${s.inserted} rows inserted, ${s.desperates} rows rejected in ${s.lifetime.toMinutes} minutes."
+        s"Statistics: ${s.inserted} rows inserted, ${s.uninsertable} rows rejected in ${s.lifetime.toMinutes} minutes."
       case s if (s.lifetime.toHours > 24) =>
-        s"Statistics: ${s.inserted} rows inserted, ${s.desperates} rows rejected in ${s.lifetime.toDays} days and ${s.lifetime.toHours - s.lifetime.toDays * 24} hours."
+        s"Statistics: ${s.inserted} rows inserted, ${s.uninsertable} rows rejected in ${s.lifetime.toDays} days and ${s.lifetime.toHours - s.lifetime.toDays * 24} hours."
       case s =>
-        s"Statistics: ${s.inserted} rows inserted, ${s.desperates} rows rejected in ${s.lifetime.toHours} hours."
+        s"Statistics: ${s.inserted} rows inserted, ${s.uninsertable} rows rejected in ${s.lifetime.toHours} hours."
     }
   }
 
-  /** Allocate all resources for an application */
+  /** Allocate all resources */
   def acquire[F[_]: ConcurrentEffect: Timer: Logger](
     cmd: RepeaterCli.ListenCommand
   ): Resource[F, Resources[F]] = {
     // It's a function because blocker needs to be created as Resource
     val initResources: F[Blocker => Resources[F]] = for {
-      transformed <- Config.transform[F](cmd.config).value
-      env         <- Sync[F].fromEither(transformed)
-      bigQuery    <- services.Database.getClient[F]
+      env      <- Sync[F].delay(cmd.env)
+      bigQuery <- services.Database.getClient[F]
+      bucket <- Sync[F].fromEither(
+        validateBucket(env.config.output.deadLetters.bucket)
+          .toEither
+          .leftMap(e => throw new RuntimeException(e.toList.mkString("\n")))
```

Comment:
The `throw` is wrong here.  You want to return the RuntimeException not throw it.